muxink/src/codec/bincode.rs

```diff
@@ -110,8 +119,87 @@ where
                 bincode::ErrorKind::Io(io_err) if io_err.kind() == io::ErrorKind::UnexpectedEof => {
                     DecodeResult::Incomplete
                 }
-                _ => DecodeResult::Failed(err),
+                bincode::ErrorKind::SizeLimit
+                | bincode::ErrorKind::SequenceMustHaveLength
+                | bincode::ErrorKind::Custom(_)
+                | bincode::ErrorKind::InvalidCharEncoding
+                | bincode::ErrorKind::InvalidTagEncoding(_)
+                | bincode::ErrorKind::DeserializeAnyNotSupported
+                | bincode::ErrorKind::Io(_)
+                | bincode::ErrorKind::InvalidUtf8Encoding(_)
+                | bincode::ErrorKind::InvalidBoolEncoding(_) => DecodeResult::Failed(err),
             },
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::DecodeResult;
+    use crate::codec::{
+        bincode::{BincodeDecoder, BincodeEncoder},
+        BytesMut, FrameDecoder, Transcoder,
+    };
+
+    #[test]
+    fn roundtrip() {
+        let data = "abc";
+
+        let mut encoder = BincodeEncoder::new();
+        let value: String = String::from(data);
+        let encoded = encoder.transcode(value).expect("should encode");
+
+        let mut decoder = BincodeDecoder::<String>::new();
+        let decoded = decoder.transcode(encoded).expect("should decode");
+
+        assert_eq!(data, decoded);
+    }
+
+    #[test]
+    fn decodes_frame() {
+        let data = b"\x03\0\0\0\0\0\0\0abc\x04\0\0\0\0\0\0\0defg";
+
+        let mut bytes: BytesMut = BytesMut::new();
+        bytes.extend(data);
+
+        let mut decoder = BincodeDecoder::<String>::new();
+
+        assert!(matches!(decoder.decode_frame(&mut bytes), DecodeResult::Item(i) if i == "abc"));
+        assert!(matches!(decoder.decode_frame(&mut bytes), DecodeResult::Item(i) if i == "defg"));
+    }
+
```

Comment:
It would be nice to add a test that reads multiple integers from a stream that has had integers directly written to it. This would be useful when/if we switch to varint encoding.

Reply:
Test added 04b601a8b98e343e30591a0a0ce258d583550c84 along with the change to varint encoding everywhere.

Reply:
> We can generalize most of these tests over any transcoder T, although this makes it quite a bit more complex
Should be much simpler with macro.